Snakefile

```diff
@@ -301,9 +301,32 @@ rule build_base_energy_totals:
         "scripts/build_base_energy_totals.py"
 
 
+# TODO: add input for default demand data
+rule prepare_default_demand_data:
+    output:
+        efficiency_gains_cagr="data/demand/efficiency_gains_cagr.csv",
+        growth_factors_cagr="data/demand/growth_factors_cagr.csv",
+        district_heating="data/demand/district_heating.csv",
+        fuel_shares="data/demand/fuel_shares.csv",
+    script:
+        "scripts/prepare_default_demand_data.py"
+
+
 rule prepare_energy_totals:
     input:
         unsd_paths="data/energy_totals_base.csv",
+        efficiency_gains_cagr="resources/custom_data/efficiency_gains_cagr.csv"
+        if config["custom_data"]["efficiency_gains_cagr"]
+        else "data/demand/efficiency_gains_cagr.csv",
+        growth_factors_cagr="resources/custom_data/growth_factors_cagr.csv"
+        if config["custom_data"]["growth_factors_cagr"]
+        else "data/demand/growth_factors_cagr.csv",
+        district_heating="resources/custom_data/district_heating.csv"
+        if config["custom_data"]["district_heating"]
+        else "data/demand/district_heating.csv",
+        fuel_shares="resources/custom_data/fuel_shares.csv"
+        if config["custom_data"]["fuel_shares"]
+        else "data/demand/fuel_shares.csv",
```

Comment:
This is very ugly.
As a suggestion, we could keep that the input data here are always in the resource folder, but we could add a rule that copies the custom_data into the appropriate resource folder.
The same applies to the other rules as well.

what do you think?

Reply:
Sounds very useful 👍 

Do you have an idea how to get the default demand data?

Reply:
I think there is a misunderstanding here. The files CAGR, fuel shares and district heating share are intended to be the default demand workflow. The user needs to make sure the country under assessment has set of values in the file ("data/demand/fuel_shares.csv")

What could be a good idea for this PR is to check if the country codes exsist in the file, if not, then apply a default set of values and add this info to the logger. An appropriate message would be "No growth factors for NA were found in data/demand/growth_factors_cagr.csv" using the default values of [set of values]"

Reply:
I would add the default values to the csv file as a row with index "DF" instead of the usual country code, and when the country code not found, we could then copy the "DF" to "NA" (in case we're working on Namibia) then log this info to the user @doneachh

Reply:
@hazemakhalek Sounds usefull! Which data would you take for the DF values?

Reply:
@doneachh I don't have a preference, but you can copy those for Morocco in the file now. You can implement it and we can adapt the values later if needed :)

Reply:
As a comment, I'd suggest to use like "DEFAULT" as flag rather than DF as it may be difficult to catch

Reply:
@davide-f sounds good to me. 
@doneachh did you start implementing the idea? If not, do you mind if I do it?
I want this to be merged maximum tomorrow, so if you don't have that in your plan, I'm happy to take it from here :)

Reply:
@hazemakhalek i already started, but i didnt finalized it. Would commit my stuff and you can go on if you want to :)

Reply:
@hazemakhalek in my opinion the only thing, what is left to do is the selection of the default data, when reading the csvs and no data is found for the key. I updated the files and changed the read_in process from hard coded paths to snakemake.input path style